### ISSUE TITLE
Support Vercel Edge environment

### ISSUE THREAD

**Issue by BrunoQuaresma** [fauna/fauna-js#173]
Hey folks, I'm developing an app using NextJS and I have a function that runs on the edge, and when I try to use the fauna client in this function, I get the following error:
```
Critical dependency: require function is used in a way in which dependencies cannot be statically extracted

Import trace for requested module:
./node_modules/fauna/dist/browser/index.js
Error [ReferenceError]: navigator is not defined
```
I guess it is because it tries to access some browser feature that is not available 🤔 

To reproduce this you create a NextJS project and inside of the app folder, create a `/api/route.ts` with something like:
```ts
export const runtime = "edge";

export function GET() {
    const fauna = new Client({ ... })
    fauna.query(fql`[]`)
}
```
Sorry for not having too much more info to help with debug 😞

**Comment by ptpaterson** [comment]
Our internal ticket for this is: [FE-3729](https://faunadb.atlassian.net/browse/FE-3729)

**Comment by ptpaterson** [comment]
The heart of the issue is that we assume all "server" environments will be Node. If it's not Node, then it must be a browser or service worker. We need to catch the neither Node nor Browser case here:

https://github.com/fauna/fauna-js/blob/5c150279bd2e808c630149d4e71a21504006b715/src/util/environment.ts#L29-L41

**Comment by cleve-fauna** [comment]
@BrunoQuaresma this is released you can pick it up in 0.9.2 should be on NPM and there's a Git Hub release as well.

**Comment by BrunoQuaresma** [comment]
Thank you, folks!
